visualization/app/codeCharta/ui/mapTreeView/mapTreeViewLevelComponent.spec.ts

```diff
@@ -0,0 +1,213 @@
+        });
+
+        it("Eye click", () => {
+            mapTreeViewLevelController.node = codeMapUtilService.getCodeMapNodeFromPath("/root/a/", "Folder");
+            mapTreeViewLevelController.codeMapActionsService.toggleNodeVisibility = jest.fn();
+            mapTreeViewLevelController.onEyeClick();
+
+            expect(mapTreeViewLevelController.codeMapActionsService.toggleNodeVisibility).toHaveBeenCalledWith(mapTreeViewLevelController.node);
+        });
+
+        it("Is leaf", () => {
+            mapTreeViewLevelController.node = codeMapUtilService.getCodeMapNodeFromPath("/root/a/ab/aba", "File");
+            expect(mapTreeViewLevelController.isLeaf()).toBeTruthy();
+        });
+
+        it("Is not leaf", () => {
+            mapTreeViewLevelController.node = codeMapUtilService.getCodeMapNodeFromPath("/root/a/ab", "Folder");
+            expect(mapTreeViewLevelController.isLeaf(mapTreeViewLevelController.node)).toBeFalsy();
+        });
+
+        it("Is blacklisted", () => {
+            mapTreeViewLevelController.node = codeMapUtilService.getCodeMapNodeFromPath("/root/a/ab", "Folder");
+
+            CodeMapUtilService.isBlacklisted = jest.fn();
+            mapTreeViewLevelController.isBlacklisted(mapTreeViewLevelController.node);
+
+            expect(CodeMapUtilService.isBlacklisted).toHaveBeenCalledWith(
+                mapTreeViewLevelController.node, settingsServiceMock.settings.blacklist, ExcludeType.exclude);
+        });
+
+        it("Not blacklisted, not exist", () => {
+            CodeMapUtilService.isBlacklisted = jest.fn();
+            let blacklisted = mapTreeViewLevelController.isBlacklisted(mapTreeViewLevelController.node);
+            expect(blacklisted).toBeFalsy();
+        });
+
+        it("Sort leaf", () => {
+            mapTreeViewLevelController.node = codeMapUtilService.getCodeMapNodeFromPath("/root/a/ab/aba", "File");
+            let sortValue = mapTreeViewLevelController.sortByFolder(mapTreeViewLevelController.node);
+            expect(sortValue).toBe(0);
+        });
+
+        it("Sort not a leaf", () => {
+            mapTreeViewLevelController.node = codeMapUtilService.getCodeMapNodeFromPath("/root/a/ab", "Folder");
+            let sortValue = mapTreeViewLevelController.sortByFolder(mapTreeViewLevelController.node);
+            expect(sortValue).toBe(1);
+        });
+
+        it("Subscribe Hover", () => {
+            let subscriber: MapTreeViewHoverEventSubscriber = {
+                onShouldHoverNode: jest.fn(),
+                onShouldUnhoverNode: jest.fn()
+            };
+
+            MapTreeViewLevelController.subscribeToHoverEvents($rootScope, subscriber);
+
+            expect($rootScope.$on).toBeCalledWith("should-hover-node", expect.any(Function));
+            expect($rootScope.$on).toBeCalledWith("should-unhover-node", expect.any(Function));
+            expect(subscriber.onShouldHoverNode).toBeCalled;
```

Comment:
These two statements always return true. It should actually be like:
`expect(subscriber.onShouldHoverNode).toBeCalled();` but these unfortunately always return false. Any ideas?

Reply:
You are right. I add a function called "whatever" to the subscriber mock and it returns true.
I think it has something to do with being inside from a anonymous function or being given as an argument into a function.
I added the line

            expect($rootScope.$on).toBeCalled();
and returns true, so toBeCalled() works

Reply:
My opinion: The problem is that subscriber is an interface.

Reply:
I completely changed the perspective